packages/foundation/src/IComponent.ts

```diff
@@ -84,7 +84,10 @@ export type IStateComponentType<TComponentProps, TViewProps> = (props: Readonly<
 /**
  * Defines the contract for view components.
  */
-export type IViewComponent<TViewProps> = (props: IPropsWithChildren<TViewProps>) => ReturnType<React.FunctionComponent>;
+export interface IViewComponent<TViewProps> {
+  (props: IPropsWithChildren<TViewProps>): ReturnType<React.FunctionComponent>;
+  name?: string;
+}
```

Comment:
Per our chat, I think this typing change is redundant and shouldn't be needed. `name` is a property of `Function` already:

![image](https://user-images.githubusercontent.com/26070760/61003078-e8fd4b00-a317-11e9-838c-eeb2314a6f11.png)